python/ambassador/ir/irhost.py

```diff
@@ -383,7 +383,8 @@ def matches_httpgroup(self, group: 'IRHTTPMappingGroup') -> bool:
         else:
             # It is NOT A TYPO that we use group.get("host") here -- whether the Mapping supplies
             # "hostname" or "host", the Mapping code normalizes to "host" internally.
-            group_glob = group.get('host') or None  # NOT A TYPO: see above.
+            redirect_glob = group.get('host_redirect', {}).get('host') if group.get('host_redirect', {}) else None
```

Comment:
I'm confused, isn't this equivalent to just
```python
redirect_glob = group.get('host_redirect', {}).get('host')
```
?

Reply:
The main issue with why this looks so gross is that `group.get('host_redirect', {})` is sometimes `None` meaning that a `group.get('host_redirect', {}).get('host')` will throw an error because you cant `NoneType.get('host')` thats why I first check if anything is returned by `group.get('host_redirect', {})` before doing `group.get('host_redirect', {}).get('host')` that said, there is still probably a better way to write that out.

Reply:
I would probably do 

```
  # It's possible for group.host_redirect to be None instead of missing, and it's also
  # conceivably possible for group.host_redirect.host to be "", which we'd rather be 
  # None. Hence we do this two-line dance to massage the various cases.
  host_redirect = group.get('host_redirect') or {}
  redirect_glob = host_redirect.get('host') or None
```

This might not be "better" but at least there's room for the explanation...